src/freshrssapi.cpp

```diff
@@ -0,0 +1,546 @@
+			"FreshRssApi::fetch_feed: Empty response: %s",
+			result);
+		return feed;
+	}
+	nlohmann::json content;
+	try {
+		content = nlohmann::json::parse(result);
+	} catch (nlohmann::json::parse_error& e) {
+		LOG(Level::ERROR,
+			"FreshRssApi::fetch_feed: reply failed to parse: %s",
+			result);
+		return feed;
+	}
+
+
+	const nlohmann::json entries = content["items"];
+	if (!entries.is_array()) {
+		LOG(Level::ERROR,
+			"FreshRssApi::fetch_feed: items is not an array");
+		return feed;
+	}
+
+	LOG(Level::DEBUG,
+		"FreshRssApi::fetch_feed: %" PRIu64 " items",
+		static_cast<uint64_t>(entries.size()));
+	try {
+		for (const auto& entry : entries) {
+			rsspp::Item item;
+
+			// Title
+			if (entry.contains("title") && !entry["title"].is_null()) {
+				item.title = entry["title"];
+			}
+
+			// Link
+			if (entry.contains("canonical") && !entry["canonical"].is_null()) {
+				for (const auto& a : entry["canonical"]) {
+					if (a.contains("href") && !a["href"].is_null()) {
+						item.link = a["href"];
+						break;
+					}
+				}
+			}
+
+			// Author
+			if (entry.contains("author") && !entry["author"].is_null()) {
+				item.author = entry["author"];
+			}
+
+			// Content
+			if (entry.contains("summary") && !entry["summary"].is_null()) {
+				for (const auto& a : entry["summary"].items()) {
+					if (!a.value().is_null()) {
+						item.content_encoded = a.value();
+						break;
+					}
+				}
+			}
+
+			// Guid
+			if (entry.contains("id") && !entry["id"].is_null()) {
+				item.guid = entry["id"];
+			}
+
+			// Publish date
+			if (entry.contains("published") && !entry["published"].is_null()) {
+				int pub_time = entry["published"];
+				time_t updated = static_cast<time_t>(pub_time);
+
+				item.pubDate = utils::mt_strf_localtime(
+						"%a, %d %b %Y %H:%M:%S %z",
+						updated);
+				item.pubDate_ts = pub_time;
+			}
+
+			// Podcast enclosure
+			if (entry.contains("enclosure") && !entry["enclosure"].is_null()) {
+				for (const auto& a : entry["enclosure"]) {
+					if (a.contains("href") && a.contains("type")
+						&& !a["href"].is_null() && !a["type"].is_null()
+						&& newsboat::utils::is_valid_podcast_type(a["type"])) {
+						item.enclosure_type = a["type"];
+						item.enclosure_url = a["href"];
+						break;
+					}
+				}
+			}
+
+			// Read/unread status
+			bool unread = true;
+			if (entry.contains("categories")
+				&& !entry["categories"].is_null()) {
+				for (const auto& a: entry["categories"]) {
+					if (a == "user/-/state/com.google/read") {
+						unread = false;
+					}
+				}
+			}
+			if (unread) {
+				item.labels.push_back("unread");
+			} else {
+				item.labels.push_back("read");
+			}
+
+			feed.items.push_back(item);
+		}
+	} catch (nlohmann::json::exception& e) {
+		LOG(Level::ERROR, "Exception occurred while parsing feed: ", e.what());
+	}
+
+	std::sort(feed.items.begin(),
+		feed.items.end(),
+	[](const rsspp::Item& a, const rsspp::Item& b) {
+		return a.pubDate_ts > b.pubDate_ts;
+	});
```

Comment:
Suggestion: is this necessary? I see the same code in other integrations, but ItemList already knows how to sort items, so this seems superfluous.

Reply:
Not needed - removed